### DISCUSSION TITLE
Event fired during sorting?

### DISCUSSION THREAD

**Discussion by akulmehta** [Power-Components/livewire-powergrid#585] [Q&A]
Hello is there an event fired during sorting of a column? I would like to run some js when the sort is performed so I need to emit a livewire event when a sort is performed. How can I achieve this?

**Comment by luanfreitasdev** [comment]
Hello!. I think you can use `updatedSortField`() method and call `dispatchBrowserEvent`

**Reply by akulmehta** [reply]
@luanfreitasdev this is not working. What was working was overriding the sortBy method (shown below). However, doing this feels risky in many tables. Is there a more elegant solution?

```
    public function sortBy(string $field, string $direction = 'asc'): void
    {
        $this->sortDirection = $this->sortField === $field ? $this->reverseSort() : $direction;

        $this->sortField = $field;

        $this->emit('updatedPostsTable'); // triggers
    }
```
